Automattic/wp-calypso
# Add update review action

Fixes https://github.com/Automattic/dotcom-forge/issues/3921

## Proposed Changes

 - add editing state to review item
 - add state to store updated review content
 - call update endpoint with the updated rating/content

## Known issues

The API currently serves the reviews with HTML formatting. We need to restrict the reviews to be plain-text in a follow-up.

## Testing Instructions


- add a review to WooCommerce Bookings by going to [http://calypso.localhost:3000/marketplace/test/[siteId]](http://calypso.localhost:3000/marketplace/test/%5BsiteId%5D)
- go to `/plugins/woocommerce-bookings`
- click read all reviews, then the `Edit my review` button
- check the styling of cancel / save buttons
- disable the `marketplace-reviews-notifications` feature flag by adding the `?flags=-marketplace-reviews-notifications` parameter to the URL
- reload the page and check that the notify checkbox is hidden
  ![CleanShot 2024-01-03 at 15 49 06@2x](https://github.com/Automattic/wp-calypso/assets/11555574/e7ac269f-e276-42bf-992b-d62cbd4a0e9b)
- update the rating and the review text and click `Save my review`
- Check that the editing view is closed and the review is updated in the list


![CleanShot 2023-12-29 at 12 56 26](https://github.com/Automattic/wp-calypso/assets/11555574/a481215a-1aec-43d4-a94f-7e2ac7ba2a09)

## Pre-merge Checklist
